### ISSUE TITLE
java Optional missing at ResponseAdapter when use union  type

### ISSUE THREAD

**Issue by doucheng** [apollographql/apollo-kotlin#4529]
### Summary

java Optional missing at ResponseAdapter when use union  type

### Version
3.7.0-SNAPSHOT

### Logs

<img width="720" alt="image" src="https://user-images.githubusercontent.com/5676888/203686746-894b14db-099d-49e8-bbc4-d1d48e213598.png">

### Steps to reproduce the behavior
*Server Schema*
<img width="720" alt="image" src="https://user-images.githubusercontent.com/5676888/203686746-894b14db-099d-49e8-bbc4-d1d48e213598.png">

*Client Operation*
```graphql
query userProfile2{
    userProfile2{
        ... on User{
            id
            name
        }
        ... on IllegalArgumentError{
            message
        }
    }
}
```
*Error*
<img width="1353" alt="image" src="https://user-images.githubusercontent.com/5676888/203687486-8452818f-4726-4902-bb8b-f5830788bc09.png">

**Comment by BoD** [comment]
Thanks a lot for spotting and reporting this! 🙏 
A fix is ongoing [here](https://github.com/apollographql/apollo-kotlin/pull/4533).

**Comment by doucheng** [comment]
Great~

**Comment by BoD** [comment]
The fix has been merged, and will be in the next release. In the meantime it will be available in the [snapshots](https://github.com/apollographql/apollo-kotlin#snapshots).

**Comment by BoD** [comment]
Fix available in v3.7.2.
